### PR TITLE
Highlight current tab on scroll in table of content in navigation

### LINKED ISSUE #5525: [Docs] Highlight current tab on scroll in table of content in navigation
#### Current State
In the table of navigation, on the right side, the current context is not highlighted.
image

#### Desired State
It should highlight the current tab context and change accordingly on scroll.
See the example here https://istio.io/latest/docs/tasks/traffic-management/egress/http-proxy/

---

#### [Contributor Guides](https://docs.meshery.io/project/contributing) and Resources
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by manikmmalhotra**
Signed-off-by: manikmmalhotra <manikmalhotra2113@gmail.com>

**Description**
In the table of navigation, on the right side, the current context is now highlighting while scrolling
Screenshot from 2022-07-06 04-46-09
This PR fixes #5525 

**Notes for Reviewers**
Always open for suggestions and improvements. Please point me out if anything should be added/corrected.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 926c7f50cb2606ac8b093e176e4107e06877ce0e at: https://62c4c7d7708e253f3c3e4d70--meshery-docs-preview.netlify.app

**Review by leecalcote — CHANGES_REQUESTED** [review]
Thanks, @manikmmalhotra!

An item that might have gone unnoticed: the last section header never receives highlight. Try clicking on "Related Reading" on this page as an example - https://62c4c7d7708e253f3c3e4d70--meshery-docs-preview.netlify.app/functionality/pattern-management#related-reading

**Comment by manikmmalhotra** [comment]
> Thanks, @manikmmalhotra!
> 
> An item that might have gone unnoticed: the last section header never receives highlight. Try clicking on "Related Reading" on this page as an example - https://62c4c7d7708e253f3c3e4d70--meshery-docs-preview.netlify.app/functionality/pattern-management#related-reading

@leecalcote i've seen that issue but it was technically correct as the last part never hit the top which normally is the case for refrencing to sidemenu, i would change that i wasn't sure about that so i left that, i will correct that and update this

**Comment by leecalcote** [comment]
You're right about it being technically accurate, which makes sense to web programmers. Less so to users. Thanks for considering this feedback.

**Comment by manikmmalhotra** [comment]
> Thanks, @manikmmalhotra!
> 
> An item that might have gone unnoticed: the last section header never receives highlight. Try clicking on "Related Reading" on this page as an example - https://62c4c7d7708e253f3c3e4d70--meshery-docs-preview.netlify.app/functionality/pattern-management#related-reading

@leecalcote fixed the bug for highlighting the last child while scrolling and updated the PR

**Comment by l5io** [comment]
🚀 Preview for commit e73af77d3056d47b3f2e52ad43c9c46cb0c4037f at: https://62c6040710d0311eddbcad0b--meshery-docs-preview.netlify.app

File: docs/_includes/toc.html [hunk 1]
```diff
@@ -33,5 +33,25 @@
...
+        $(window).scroll(function(e){
+          // Detect how far are we from the top of the page
+          let windowTop = $(this).scrollTop();
+          // Loop through every navigation menu item
+          $('.md-nav__link').each(function (event) {       
+              if (windowTop >= $($(this).attr('href')).offset().top - 100) {
+              // Remove 'selected' from previously highlighted menu items
+              $('.selected').removeClass('selected');
+              // Highlight the current menu item by adding 'selected' class
+              $(this).addClass('selected');
```
Comment (adithyaakrishna) [inline docs/_includes/toc.html:48]: Loved the way you have added comments, huge ++ @manikmmalhotra

**Review by adithyaakrishna — APPROVED** [review]

**Review by warunicorn19 — COMMENTED** [review]
Screenshot 2022-07-09 at 1 40 19 AM
With the javascript turned off, the table of contents disappears

**Comment by leecalcote** [comment]
Thanks, @manikmmalhotra! 🎈

**Pull request merged**